### PR TITLE
[testing] [local] optionally skip connectivity tests

### PR DESCRIPTION
**Pull request by nammn**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

## Changes
- only log connection success if attempts were more than 1, otherwise its just noisy
- add a make target to not cleanup the environment, useful for debugging broken e2e tests
- adds `options` to our go test make targets i.e. ` make e2e-telepresence test=replica_set cleanup=false option=-short`
- adds a test skip option in our testing connectivity suite. Albeit we already have the function `SkipTestIfLocal` we would need a bit more of a refactor to properly support it. I thought adding this check is small enough to not warrant the method refactor

```
 > make e2e-telepresence test=replica_set cleanup=false options=-short
....
=== RUN   TestReplicaSet/AutomationConfig_has_the_correct_version
--- PASS: TestReplicaSet (189.24s)
    --- PASS: TestReplicaSet/Create_MongoDB_Resource (0.01s)
    --- PASS: TestReplicaSet/Basic_tests (125.03s)
        --- PASS: TestReplicaSet/Basic_tests/Secret_Was_Correctly_Created (5.00s)
        --- PASS: TestReplicaSet/Basic_tests/Stateful_Set_Reaches_Ready_State (105.02s)
        --- PASS: TestReplicaSet/Basic_tests/MongoDB_Reaches_Running_Phase (15.00s)
        --- PASS: TestReplicaSet/Basic_tests/Stateful_Set_Has_OwnerReference (0.00s)
        --- PASS: TestReplicaSet/Basic_tests/Service_Set_Has_OwnerReference (0.00s)
        --- PASS: TestReplicaSet/Basic_tests/Agent_Secrets_Have_OwnerReference (0.00s)
        --- PASS: TestReplicaSet/Basic_tests/Connection_string_secrets_are_configured (0.00s)
    --- PASS: TestReplicaSet/Keyfile_authentication_is_configured (10.04s)
    --- SKIP: TestReplicaSet/Test_Basic_Connectivity (0.00s)
    --- SKIP: TestReplicaSet/Test_SRV_Connectivity (0.00s)
    --- SKIP: TestReplicaSet/Test_Basic_Connectivity_with_generated_connection_string_secret (0.00s)
    --- SKIP: TestReplicaSet/Test_SRV_Connectivity_with_generated_connection_string_secret (0.00s)
    --- PASS: TestReplicaSet/Ensure_Authentication (20.05s)
        --- PASS: TestReplicaSet/Ensure_Authentication/Ensure_keyFile_authentication_is_configured (10.02s)
        --- PASS: TestReplicaSet/Ensure_Authentication/SCRAM-SHA-256_is_configured (10.03s)
    --- PASS: TestReplicaSet/AutomationConfig_has_the_correct_version (0.00s)
PASS
ok      github.com/mongodb/mongodb-kubernetes-operator/test/e2e/replica_set     189.792s
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -119,15 +119,18 @@ manifests: controller-gen ## Generate manifests e.g. CRD, RBAC etc.
 e2e-telepresence: cleanup-e2e install ## Run e2e tests locally using go build while also setting up a proxy e.g. make e2e-telepresence test=replica_set cleanup=true
 	telepresence connect; \
 	eval $$(scripts/dev/get_e2e_env_vars.py $(cleanup)); \
-	go test -v -timeout=30m -failfast ./test/e2e/$(test); \
+	go test -v -timeout=30m -failfast $(options) ./test/e2e/$(test) ; \
 	telepresence quit
 
-e2e-k8s: cleanup-e2e install e2e-image ## Run e2e test by deploying test image in kubernetes.
-	python scripts/dev/e2e.py --perform-cleanup --test $(test)
+e2e-k8s: cleanup-e2e install e2e-image ## Run e2e test by deploying test image in kubernetes without cleanup of the resources.
```
Comment (nammn) [inline Makefile:125]: grepping shows that we don't use this target in our workflow anywhere, making this change most likely safe.

Comment (mircea-cosbuc) [reply Makefile:125]: Can we make this target take additional arguments instead? (since `e2e.py` supports additional flags which we might want to use)

**Review by mircea-cosbuc — APPROVED** [review]
LGTM with one comment

File: Makefile (hunk 1 above)
Comment (nammn) [reply Makefile]: added in: 41ec97481ed2b31e3d59f1105994cd902a45b855

**Pull request merged**